app/controllers/students_controller.rb

```diff
@@ -30,6 +29,11 @@ def show
     }
   end
 
+  def restricted_notes
+    raise Exceptions::EducatorNotAuthorized unless current_educator.can_view_restricted_notes
```

Comment:
Does `authorize!` not fire before every action here by default, except `names`? line 6, `before_action :authorize!, except: [:names]`.